DOCS-6157-added-links (#8)
mulesoft/docs-service-mesh: modules/ROOT/pages/configure-service-mesh.adoc

> 

Reviewer: HIDDEN TEXT:

= Configure Service Mesh
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: service mesh, microservices, downloading, installing
image::configure-service-mesh-breadcrumb.png[]

> The following topics are covered in this section:

Reviewer: GLOBALLY remove self-referential text (the following topics, this section) and replace them with...:

     Direct user address
     WHY user cares
     Summary of content

"To configure Anypoint Service Mesh, you must first understand and provision the Service Mesh adapter, create a non-Mule API, and bind the adapter to that API."

Author: Done!

Author: Done!

* <<adapter-overview,Service Mesh Overview>>
* <<provision-adapter,Provision Service Mesh Adapter>>
* <<create-non-mule-app,Create a non-Mule application>>

> * <<bind-adapter,Bind Service Mesh Adapter>>

Reviewer: OMIT lists like this. The audience can see the TOC to the left and other navigation to the right of the text screen.
We don't want to use self-referential text, and we want to minimize active linking within the narrative. GLOBAL

Author: Done!

Author: Done! Thanks!

[[adapter-overview]]
== Service Mesh Adapter Overview

> After you install Service Mesh, you must provision the Service Mesh adapter and bind it with a specific namespace. When you provision the adapter, you must specify the size of the adapter based on the number of CPUs and memory of the system that will be managed in Service Mesh.

Reviewer: Why does the intro text to this topic address provisioning and binding, which are subsequent topics?
This topic is supposed to be an overview to the adapter; as such, it should address WHAT the adapter is and why that matters to the user. It should also prepare the audience to read the enclosed subtopics: adapter plan types, adapter status, and CLI help for adapter commands. 

I suggest replacing the previous two sentences with text that covers these things instead.

Should you keep the existing text, implement this edit:

```suggestion
After you install Anypoint Service Mesh, provision the Service Mesh adapter and bind it with a specific namespace. When you provision the adapter, specify the size of the adapter based on the number of CPUs and memory of the system managed in Service Mesh.
```

Author: Done!

> After you install Service Mesh, you must provision the Service Mesh adapter and bind it with a specific namespace. When you provision the adapter, you must specify the size of the adapter based on the number of CPUs and memory of the system that will be managed in Service Mesh.

Reviewer: HIDDEN TEXT: SEE EDITS
Note: use backticks to represent syntax values, rather than quotation marks.

=== Adapter Plan

Anypoint Service Mesh currently supports small, medium, and large adapter plans:

[%header%autowidth.spread]
|===
| Plan | CPU (cores) | Memory (GiB) | API Limit
| Small | 500 | 1 | 25
| Medium | 1 | 1.5 | 50
| Large | 2 | 2 | 100
|===

You can also modify the size of an adapter after it has been provisioned. If you update the plan to change the plan size or any other parameter, the adapter deployment is updated appropriately. Kubernetes then creates a pod with new information. However, the old adapter continues handling requests until the new pod is ready.

After the new adapter is ready, the old pod is terminated and all requests are routed to the new adapter. If the update fails due to insufficient space to start up a new pod, or if the pod is not ready, an error message is recorded in a Kubernetes event. In such a scenario, the old adapter continues to handle requests because the new pod is not yet available.

=== Adapter Status

After you execute an adapter command to provision or bind an adapter, a status displays, indicating the success or failure of the command:

* Ready
The adapter is ready to be used.
* Failed
The adapter was not created successfully and an error displays. For more information about the error, inspect the adapter logs by using the command ./service-mesh adapter logs --name=<name> --namespace=<namespace>.
* Provisioning
The adapter is in the process of being provisioned.
* Binding
The adapter is in the process of being associated with a specific application.

In the following example, a binding called `customers` is created to bind the `anypoint-service-mesh-adapter` adapter with the `customers` service. This binding is created in the `nto-payments` namespace, which is the same namespace in which the adapter was previously created, and it is associated with the API that has the ID `1234`:

[source,text,linenums]
NAMESPACE NAME ADAPTER API ID SERVICE NAME STATUS
nto-payments customers anypoint-service-mesh-adapter 1234 customers Ready

=== CLI Help for Adapter Commands

After you install Service Mesh, use the command line interface (CLI) to help you use appropriate commands:
+
./service-mesh help

image::service-mesh-cli-adapter-help.png[CLI Help for Adapter Commands]

Author: Done!

image::service-mesh-cli-adapter-help.png[CLI Help for Adapter Commands]

[[provision-adapter]]
== Provision Service Mesh Adapter

> Before	you can start managing your API using Service Mesh, you must prepare your namespace by provisioning a Service Mesh Adapter. Provisioning prepares the namespace for the Service Mesh configuration. After you provision the adapter, you must redeploy your existing application to ensure that the Envoy sidecar is injected with each pod.

Reviewer: Service Mesh adapter or Service Mesh Adapter? 
Search for and standardize GLOBALLY.

Reviewer: If this topic focus is provisioning, why mention anything "after you provision the adapter"?
Also, will your audience know what "the Envoy sidecar" is, injection, and pod?
Suggest omitting.
I also suggest omitting the second sentence, which adds nothing to the first.
```suggestion
Before you can start managing your API using Service Mesh, you must prepare your namespace by provisioning a Service Mesh adapter:
```

Author: Kept the sentence as is. Content needed.

> To provision the Anypoint Service Mesh adapter:

Reviewer: OMIT in favor of the previous sentence ending with a colon.

Reviewer: HIDDEN TEXT
Unedited due to time constraints (and the likelihood of change)

. Determine the adapter size based on the plan.
. From the Service Mesh installer download location in the Command window, type the following command to create the adapter:
+
[source,text,linenums]
----
./service-mesh adapter create 
--name=<adapter name>
--namespace=<adapter namespace>
--size=<adapter plan size>
--clientId=<clientId of the application> 
--clientSecret=<clientSecret of the application> 
--platformUri=<URL of Anypoint Platform>
----
+
[%header%autowidth.spread]
|===
| Parameter Name | Description | Required/Optional | Default Value
| name | The name of the adapter that you want to create, for example, “service-mesh-test-adapter”. | Required | N/A
| namespace | The namespace in which you want to create the adapter, for example, “mulesoft-apps”. | Optional | default
| size | The size of the adapter based on the number of CPUs and memory to use | Optional | Medium
| clientId | The client ID of the application | Required | N/A
| clientSecret | The client secret of the application | Required | N/A
| platformUri | The URL of the host where Anypoint Platform is installed | Optional | https://anypoint.mulesoft.com
|===
+
. Verify the adapter status:
+   
`./service-mesh adapter list`
+
image::provision-service-mesh-adapter-output.png[Example Output]
+
. Review adapter logs to see detailed information about Kubernetes events and adapter transactions:
+
`./service-mesh adapter logs --name=<adapter_name> --namespace=<namespace>`

+
`./service-mesh adapter logs --name=<adapter_name> --namespace=<namespace>`

[[create-non-mule-app]]

> == Create a non-Mule API

Reviewer: Why a NON Mule API? Explain.

Reviewer: Also...do we really have to cover how to create an API here? Shouldn't having one already be a prerequisite to using Service Mesh?

> If your MuleSoft or non-MuleSoft application does not exist, you must create it before you bind the service to the adapter. 

Reviewer: ```suggestion
You must have an application before you can bind the Service Mesh adapter to it.
```
Is this technically correct? I am confused because we seem to be using more than one word for API/namespace/service? GLOBALLY select the most precise term possible and use that one term consistently throughout a deliverable. Avoid synonyms.

Reviewer: HIDDEN TEXT
unedited

To create an API, perform the following steps:
. Go to *Anypoint Platform > API Manager* and select the *Manage API* drop-down list. 
. Select *Create new API*.
+
The Creating an asset page is displayed.
. Specify the *Name*, *Asset types* details for the API.  
+
The Endpoint Confoguration window id displayed, as shown in the following screenshot.
+
image::create-non-mule-api-service-mesh.png[85%, 85%]
+
. Specify the required details on the page and click *Save*.
+
Your API is now created.

Author: Done!

+
Your API is now created.

[[bind-adapter]]

> == Bind your service to an API on API Manager

Reviewer: What is "your service"? The user created an API and provisioned an adapter, but a service...?

If you keep this title:
```suggestion
== Bind Your Service to an API on API Manager
```

> After you provision the adapter, you must bind your service with an existing API to the adapter that you created. The adapter uses the unique API ID of the API for this binding. The Anypoint Platform uses the API ID to identify an application and maintains its configuration. The adapter uses this API ID to fetch policies from the Anypoint Platform and applies to your services that are running in the cluster.

Reviewer: ```suggestion
After you provision the adapter, you must bind your service to it, using an existing API. The adapter uses the unique ID of the API for this binding, fetching policies from Anypoint Platform and applying them to your services that are running in the cluster.
```
Is this still technically accurate?
Avoid repetition and wordiness.

Author: Done!

> To bind the service with an API, perform the following steps:

Reviewer: What service? Your service? 
Which API? Your API?

> . Obtain the API ID of the API that you want to bind with the service.

Reviewer: Wouldn't the user already have this, based on the previous text? 
If you keep this step:
```suggestion
. Obtain the ID of the API that you want to bind with the service.
```

> . If your applications were already running in the namespace, restart your application to inject the sidecar in the pod:

Reviewer: ```suggestion
. If your application is already running, restart it to inject the sidecar in the pod, specifying the following syntax with details specific to your implementation:
```
Is the application likely to be running already?
Does your audience understand what a sidecar and pod are?

+
`$ kubectl -n <namespace> patch deploy <deployname> --type=json -p='[{"op": "replace", "path": "/spec/template/metadata/labels/service-mesh.mulesoft.com","value":"enable"}]`
+

> In the above command, replace the namespace and the deploy name with your specific details.

Reviewer: OMIT per inclusion in the intro text.

+

> . Verify that the application Kubernetes service port is named following the requirements requested by Istio: https://istio.io/docs/setup/additional-setup/requirements/.

Reviewer: Do we need all of "the application Kubernetes service port"? How about just "service port"?
Will your audience know what Istio is?

> . From the Service Mesh installer download location in the Command window, type the following command to bind the adapter with an application:

Reviewer: "an application" or "your application"? Standardize: are we giving the audience an example using its values or just an example of any situation?

Reviewer: HIDDEN TEXT
unedited

----
./service-mesh binding create 
--name=<servicename>
--namespace=<namespace name>
--adapter=<adapter name>
--apiId=<application ID> 
--serviceName=<servicename>
----

+
[%header%autowidth.spread]
|===
| Parameter Name | Description | Required/Optional | Default Value
| name | The name that identifies the binding that you want to create, for example, service-binding-anypoint-security”. 
 | Required | N/A
| namespace | The Kubernetes namespace in which you want to create the binding, for example, “mulesoft-apps”. Make sure that you use the same namespace as the adapter you provisioned in the previous procedure. | Required | N/A
| adapter | Specify the name of the adapter that you provisioned in the previous procedure.  | Required | N/A 
| apiId | Specify the API ID of the API that you want to associate with this adapter. | Required | N/A.
 Use the API ID you obtained as part of installation prerequisites as explained in this document.
| serviceName | The name of the service to be binded to the API, which will be managed by Anypoint Platform. | Required | N/A.
Use the client ID and client secret you obtained as part of installation prerequisites as explained in this document.
|===
+

. Verify the binding status using the following command:
`./service-mesh binding list`
+

After the command completes processing, the binding becomes activate within a few seconds and the API is successfully registered to the proxy. An output is generated, as shown in the following example:
+

image::service-mesh-adapter-binding.png[Example Output]
+ 
After you bind your API, the API in API Manager is marked as Active. You are now ready to apply policies to protect your APIs.
+

image::api-manager-status-for-service-mesh.png[85%, 85%]

== See Also
* xref:api-manager::find-api-id-task.adoc[Obtain API ID]
